src/validation.cpp

```diff
@@ -5157,7 +5157,10 @@ void ChainstateManager::LoadExternalBlockFile(
                 }
 
                 // Activate the genesis block so normal node progress can continue
-                if (hash == params.GetConsensus().hashGenesisBlock) {
+                // Do this only if genesis isn't activated yet, to avoid connecting many blocks
+                // without assumevalid in the case of a continuation of a reindex that
+                // was interrupted by the user.
+                if (hash == params.GetConsensus().hashGenesisBlock && WITH_LOCK(::cs_main, return ActiveHeight()) == -1) {
```

Comment:
In commit "validation: fix issue with an interrupted -reindex" (570af7281e21d51852b706e891835758d3d7161f)

It would be helpful if comment below mentioned that ActivateBestChain call will only connect the genesis block, not other blocks because ActivateBestChain on connects blocks which are in the block tree db, and block tree db only contains blocks whose parents are in it.

Otherwise it's not clear why ActivateBestChain is being called to connect just one block and why it wouldn't connect other blocks, or at least it wasn't clear to me.

Alternately if there's are some cases where this check is imperfect and ActivateBestChain could connect extra blocks, it would be good if comment clarified to say that intentions of this code is only to connect genesis block but there are case it could connect other blocks.

Reply:
I extended the comment a bit.
I think one more theoretical case in which `ActivateBestChain` could have connected extra blocks (but wouldn't anymore after this PR) is  if, for some reason, block files loaded here (whether with `-reindex` or `-loadblock`) would contain the genesis block multiple times.